### PR TITLE
Adds benchmarks for nallocx, calloc, alloc, alloc_excess, ralloc, ralloc_excess

### PR DESCRIPTION
**Pull request by gnzlbg**

### CODE REVIEW

**Comment by gnzlbg** [comment]
@alexcrichton I am adding these benchmarks to jemallocator even though they only use `jemalloc-sys`, I hope that is ok.

**Comment by SimonSapin** [comment]
> Its a lot of boilerplate that I don't seem to be able to avoid because I cannot concatenate identifiers to form new identifiers in macros.

Rather than generating functions whose names are concatenated identifiers, I’ve had some success making macros generate functions in modules. So instead of `${foo}_${bar}` you end up with `$foo::$bar` for example.

**Comment by gnzlbg** [comment]
@SimonSapin the problem I ran into was that I had the `size` and `align` as part of the test name. I would have to move them to be part of the module's name, but I don't think that is possible.

File: jemalloc-sys/build.rs [hunk 1]
```diff
@@ -39,8 +49,23 @@ fn main() {
...
+        let mut git_cmd = Command::new("git");
+        git_cmd.arg("submodule")
+            .arg("update")
+            .arg("--init")
+            .arg("--recursive");
+        run (&mut git_cmd);
+    }
+    if !configure.exists() {
+        panic!("jemalloc/configure does not exist");
+    }
```
Comment (alexcrichton) [inline jemalloc-sys/build.rs]: Can this be left out to a future PR?

The submodule update here I don't think shoudl be run with `run` which asserts success, it should just be opportunistic and should provide a better failure message unrelated to git if the git step ends up failing.

Comment (gnzlbg) [reply jemalloc-sys/build.rs]: @alexcrichton the submodule update is a mistake in this PR, i have reverted it. 

I have it in a separate PR, but `git add -u` pulled into all my other PRs making `i686-apple` builds fail... Sorry about that.

**Pull request merged**